File: momentra/momentra/Design/Shell/MomentraShellTheme.swift
import SwiftUI

enum GlobalTheme {
    static let topBarBackground = Color(hex: 0x0C0F15)
    static let bottomBarBackground = Color(hex: 0x0C0F15)
    static let surfaceContent = Color(hex: 0x14121B)
    static let contextUnselected = Color(hex: 0xC9C4D9)
    static let bottomSelected = Color(hex: 0xC9BFFF)
    static let bottomUnselected = Color(hex: 0xC9C4D8)
    static let companyChipBackground = Color(hex: 0x1A2030)
    static let companyChipBorder = Color(hex: 0x3A4258)
    static let actionCircle = Color(hex: 0x1E293B)
    static let statusOnline = Color(hex: 0x10B981)
    static let moduleCardBackground = Color(hex: 0x161B26)
    static let createMomentCta = Color(hex: 0xE8621A)
}

struct ContextTheme {
    let context: AppContextKind
    let contextAccent: Color
    let contextAccentSecondary: Color

    static func of(_ context: AppContextKind) -> ContextTheme {
        switch context {
        case .personal:
            return ContextTheme(context: .personal, contextAccent: Color(hex: 0x7C5CFC), contextAccentSecondary: Color(hex: 0xA78BFA))
        case .group:
            return ContextTheme(context: .group, contextAccent: Color(hex: 0xE8621A), contextAccentSecondary: Color(hex: 0xFF8E63))
        case .business:
            return ContextTheme(context: .business, contextAccent: Color(hex: 0x818CF8), contextAccentSecondary: Color(hex: 0xA5B4FC))
        case .circle:
            return ContextTheme(context: .circle, contextAccent: Color(hex: 0xE86BA3), contextAccentSecondary: Color(hex: 0xFF6B8A))
        }
    }
}

/// Cross-context global surface — NOT a selectable AppContext. Figma Coming Soon `1075:7637`.
enum GlobalSurfaceTheme {
    struct Life360 {
        let surface = GlobalTheme.topBarBackground
        let action = GlobalTheme.actionCircle
        let online = GlobalTheme.statusOnline
        /// Coming Soon page background — Figma `#14121B`.
        let comingSoonBackground = Color(hex: 0x14121B)
        let card = Color(hex: 0x161B26)
        let gold = Color(hex: 0xF2CA50)
        let goldEnd = Color(hex: 0xFFAB40)
        let textPrimary = Color(hex: 0xE5E0EE)
        let textSecondary = Color(hex: 0xC9C4D8)
        let decorativeProgressFraction: CGFloat = 0.65
    }

    static let life360 = Life360()
}

/// Circle context Coming Soon surface — Figma `1075:7556`.
enum CircleComingSoonTheme {
    static let pageStart = Color(hex: 0x14121B)
    static let pageEnd = Color(hex: 0x1C1B1B)
    static let card = Color(hex: 0x161B26)
    static let cardAlt = Color(hex: 0x1C1B1B)
    static let accent = Color(hex: 0xE86BA3)
    static let accentEnd = Color(hex: 0xFF6B8A)
    static let lavender = Color(hex: 0xB794F6)
    static let peach = Color(hex: 0xFFB5A7)
    static let textPrimary = Color(hex: 0xE5E2E1)
    static let textSecondary = Color(hex: 0xD0C5AF).opacity(0.8)
    static let selectedTab = Color(hex: 0xFC6A8B)
    static let decorativeProgressFraction: CGFloat = 0.45
}

struct MomentTheme {
    let family: String
    let type: String
    let primary: Color
    let secondary: Color
    let surfaceTint: Color
    let icon: Color
}

enum MomentThemes {
    static func resolve(context: AppContextKind, momentTypeCode: String?) -> MomentTheme {
        switch context {
        case .personal: return personal(momentTypeCode)
        case .group: return group(momentTypeCode)
        case .business: return business(momentTypeCode)
        case .circle: return personal(nil)
        }
    }

    static func personal(_ momentTypeCode: String?) -> MomentTheme {
        let code = momentTypeCode?.uppercased() ?? ""
        if code == "FUTURE_GOAL" || code == "FUTURE_BUILDING" {
            return base("PERSONAL", "FUTURE_BUILDING", Color(hex: 0x10B981), Color(hex: 0x34D399))
        }
        if code == "LIFESTYLE" {
            return base("PERSONAL", "LIFESTYLE", Color(hex: 0x0EA5A4), Color(hex: 0x7C5CFC))
        }
        if code == "RELATIONSHIP_CONNECTION" || code == "RELATIONSHIPS" {
            return base("PERSONAL", "RELATIONSHIPS", Color(hex: 0xE91E63), Color(hex: 0xE12A9E))
        }
        return base("PERSONAL", "LIFE_OPERATIONS", Color(hex: 0x7C5CFC), Color(hex: 0xA78BFA))
    }

    static func group(_ momentTypeCode: String?) -> MomentTheme {
        let code = momentTypeCode?.uppercased() ?? ""
        if code.contains("WEDDING") { return base("GROUP", "WEDDING", Color(hex: 0xEC4899), Color(hex: 0xF472B6)) }
        if code.contains("FLATMATES") { return base("GROUP", "FLATMATES", Color(hex: 0xE8744F), Color(hex: 0xFF7A3D)) }
        if code.contains("FAMILY_HOUSEHOLD") { return base("GROUP", "FAMILY_HOUSEHOLD", Color(hex: 0xEC4899), Color(hex: 0xF472B6)) }
        if code.contains("CO_LIVING") { return base("GROUP", "CO_LIVING", Color(hex: 0x3B82F6), Color(hex: 0x60A5FA)) }
        if code.contains("COMMUNITY_LIVING") { return base("GROUP", "COMMUNITY_LIVING", Color(hex: 0x14B8A6), Color(hex: 0x2DD4BF)) }
        if code.contains("HOUSE_PARTY") || code.contains("PARTY") { return base("GROUP", "HOUSE_PARTY", Color(hex: 0x3B82F6), Color(hex: 0x60A5FA)) }
        if code.contains("OUTING") || code.contains("OFFICE") { return base("GROUP", "OFFICE_OUTING", Color(hex: 0x14B8A6), Color(hex: 0x2DD4BF)) }
        if code.contains("GIFT") { return base("GROUP", "GIFT_POOL", Color(hex: 0xE8621A), Color(hex: 0xFF8E63)) }
        if code.contains("PURCHASE") || code.contains("ASSET") { return base("GROUP", "GROUP_PURCHASE", Color(hex: 0xE8621A), Color(hex: 0xFF8E63)) }
        if code.contains("TRIP") { return base("GROUP", "TRIP", Color(hex: 0xE8744F), Color(hex: 0xFF8E63)) }
        return base("GROUP", "CUSTOM", Color(hex: 0xE8621A), Color(hex: 0xC9C4D9))
    }

    static func business(_ momentTypeCode: String?) -> MomentTheme {
        let code = momentTypeCode?.uppercased() ?? ""
        if code.contains("RUNWAY") { return base("BUSINESS", "BUSINESS_RUNWAY", Color(hex: 0x34D399), Color(hex: 0x6EE7B7)) }
        if code.contains("TEAM") { return base("BUSINESS", "TEAM_OPERATIONS", Color(hex: 0x818CF8), Color(hex: 0xA5B4FC)) }
        return base("BUSINESS", "BUSINESS_OPERATIONS", Color(hex: 0x818CF8), Color(hex: 0xFB923C))
    }

    private static func base(_ family: String, _ type: String, _ primary: Color, _ secondary: Color) -> MomentTheme {
        MomentTheme(
            family: family,
            type: type,
            primary: primary,
            secondary: secondary,
            surfaceTint: primary.opacity(0.16),
            icon: primary
        )
    }
}

extension Color {
    init(hex: UInt32, alpha: Double = 1) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xFF) / 255,
            green: Double((hex >> 8) & 0xFF) / 255,
            blue: Double(hex & 0xFF) / 255,
            opacity: alpha
        )
    }
}

File: apk/app/src/main/java/com/example/momentra/ui/theme/shell/MomentraShellTheme.kt
package com.example.momentra.ui.theme.shell

import androidx.compose.ui.graphics.Color
import com.example.momentra.domain.AppContext
import com.example.momentra.ui.theme.ShellTokens

/** Global shell chrome — typography/spacing live alongside ShellTokens dimensions. */
object GlobalTheme {
    val topBarBackground: Color = ShellTokens.TopBarBackground
    val bottomBarBackground: Color = ShellTokens.BottomBarBackground
    val surfaceContent: Color = ShellTokens.SurfaceContent
    val contextUnselected: Color = ShellTokens.ContextUnselected
    val bottomSelected: Color = ShellTokens.BottomSelected
    val bottomUnselected: Color = ShellTokens.BottomUnselected
    val companyChipBackground: Color = ShellTokens.CompanyChipBackground
    val companyChipBorder: Color = ShellTokens.CompanyChipBorder
    val actionCircle: Color = ShellTokens.ActionCircle
    val statusOnline: Color = ShellTokens.StatusOnline
    val moduleCardBackground: Color = ShellTokens.ModuleCardBackground
    val createMomentCta: Color = Color(0xFFE8621A)
}

data class ContextTheme(
    val context: AppContext,
    val contextAccent: Color,
    val contextAccentSecondary: Color,
)

object ContextThemes {
    private val personal = ContextTheme(AppContext.PERSONAL, Color(0xFF7C5CFC), Color(0xFFA78BFA))
    private val group = ContextTheme(AppContext.GROUP, Color(0xFFE8621A), Color(0xFFFF8E63))
    private val business = ContextTheme(AppContext.BUSINESS, Color(0xFF818CF8), Color(0xFFA5B4FC))
    /** Figma `1075:7556` Circle Coming Soon — pink, not Personal purple. */
    private val circle = ContextTheme(AppContext.CIRCLE, Color(0xFFE86BA3), Color(0xFFFF6B8A))

    fun of(context: AppContext): ContextTheme = when (context) {
        AppContext.PERSONAL -> personal
        AppContext.GROUP -> group
        AppContext.BUSINESS -> business
        AppContext.CIRCLE -> circle
    }
}

/** Cross-context global surface — NOT a selectable AppContext. Figma Coming Soon `1075:7637`. */
object GlobalSurfaceTheme {
    data class Life360(
        val surface: Color = GlobalTheme.topBarBackground,
        val action: Color = GlobalTheme.actionCircle,
        val online: Color = GlobalTheme.statusOnline,
        /** Coming Soon page background — Figma `#14121B`. */
        val comingSoonBackground: Color = Color(0xFF14121B),
        val card: Color = Color(0xFF161B26),
        val gold: Color = Color(0xFFF2CA50),
        val goldEnd: Color = Color(0xFFFFAB40),
        val textPrimary: Color = Color(0xFFE5E0EE),
        val textSecondary: Color = Color(0xFFC9C4D8),
        val decorativeProgressFraction: Float = 0.65f,
    )

    val life360 = Life360()
}

/** Circle context Coming Soon surface — Figma `1075:7556`. */
object CircleComingSoonTheme {
    val pageStart = Color(0xFF14121B)
    val pageEnd = Color(0xFF1C1B1B)
    val card = Color(0xFF161B26)
    val cardAlt = Color(0xFF1C1B1B)
    val accent = Color(0xFFE86BA3)
    val accentEnd = Color(0xFFFF6B8A)
    val lavender = Color(0xFFB794F6)
    val peach = Color(0xFFFFB5A7)
    val textPrimary = Color(0xFFE5E2E1)
    val textSecondary = Color(0xCCD0C5AF) // ~80% of #D0C5AF
    val selectedTab = Color(0xFFFC6A8B)
    const val decorativeProgressFraction = 0.45f
}

data class MomentTheme(
    val family: String,
    val type: String,
    val primary: Color,
    val secondary: Color,
    val surfaceTint: Color,
    val icon: Color,
)

object MomentThemes {
    fun personal(momentTypeCode: String?): MomentTheme {
        val code = momentTypeCode?.uppercase().orEmpty()
        return when {
            code == "FUTURE_GOAL" || code == "FUTURE_BUILDING" -> MomentTheme(
                family = "PERSONAL",
                type = "FUTURE_BUILDING",
                primary = Color(0xFF10B981),
                secondary = Color(0xFF34D399),
                surfaceTint = Color(0xFF10B981).copy(alpha = 0.16f),
                icon = Color(0xFF10B981),
            )
            code == "LIFESTYLE" -> MomentTheme(
                family = "PERSONAL",
                type = "LIFESTYLE",
                primary = Color(0xFF0EA5A4),
                secondary = Color(0xFF7C5CFC),
                surfaceTint = Color(0xFF0EA5A4).copy(alpha = 0.16f),
                icon = Color(0xFF0EA5A4),
            )
            code == "RELATIONSHIP_CONNECTION" || code == "RELATIONSHIPS" -> MomentTheme(
                family = "PERSONAL",
                type = "RELATIONSHIPS",
                primary = Color(0xFFE91E63),
                secondary = Color(0xFFE12A9E),
                surfaceTint = Color(0xFFE91E63).copy(alpha = 0.16f),
                icon = Color(0xFFE91E63),
            )
            else -> MomentTheme(
                family = "PERSONAL",
                type = "LIFE_OPERATIONS",
                primary = Color(0xFF7C5CFC),
                secondary = Color(0xFFA78BFA),
                surfaceTint = Color(0xFF7C5CFC).copy(alpha = 0.16f),
                icon = Color(0xFF7C5CFC),
            )
        }
    }

    fun group(momentTypeCode: String?): MomentTheme {
        val code = momentTypeCode?.uppercase().orEmpty()
        return when {
            code.contains("WEDDING") -> base("GROUP", "WEDDING", Color(0xFFEC4899), Color(0xFFF472B6))
            code.contains("PARTY") || code.contains("HOUSE") -> base("GROUP", "HOUSE_PARTY", Color(0xFF3B82F6), Color(0xFF60A5FA))
            code.contains("OUTING") || code.contains("OFFICE") -> base("GROUP", "OFFICE_OUTING", Color(0xFF14B8A6), Color(0xFF2DD4BF))
            code.contains("GIFT") -> base("GROUP", "GIFT_POOL", Color(0xFFE8621A), Color(0xFFFF8E63))
            code.contains("PURCHASE") || code.contains("ASSET") -> base("GROUP", "GROUP_PURCHASE", Color(0xFFE8621A), Color(0xFFFF8E63))
            code.contains("FLAT") || code.contains("CO_LIVING") || code.contains("COLIVING") ->
                base("GROUP", "CO_LIVING", Color(0xFF3B82F6), Color(0xFF60A5FA))
            code.contains("HOUSEHOLD") || code.contains("FAMILY") ->
                base("GROUP", "FAMILY_HOUSEHOLD", Color(0xFFEC4899), Color(0xFFF472B6))
            code.contains("TRIP") -> base("GROUP", "TRIP", Color(0xFFE8744F), Color(0xFFFF8E63))
            else -> base("GROUP", "CUSTOM", Color(0xFFE8621A), Color(0xFFC9C4D9))
        }
    }

    fun business(momentTypeCode: String?): MomentTheme {
        val code = momentTypeCode?.uppercase().orEmpty()
        return when {
            code.contains("RUNWAY") -> base("BUSINESS", "BUSINESS_RUNWAY", Color(0xFF34D399), Color(0xFF6EE7B7))
            code.contains("TEAM") -> base("BUSINESS", "TEAM_OPERATIONS", Color(0xFF818CF8), Color(0xFFA5B4FC))
            else -> base("BUSINESS", "BUSINESS_OPERATIONS", Color(0xFF818CF8), Color(0xFFFB923C))
        }
    }

    fun resolve(context: AppContext, momentTypeCode: String?): MomentTheme = when (context) {
        AppContext.PERSONAL -> personal(momentTypeCode)
        AppContext.GROUP -> group(momentTypeCode)
        AppContext.BUSINESS -> business(momentTypeCode)
        AppContext.CIRCLE -> personal(null)
    }

    private fun base(family: String, type: String, primary: Color, secondary: Color) = MomentTheme(
        family = family,
        type = type,
        primary = primary,
        secondary = secondary,
        surfaceTint = primary.copy(alpha = 0.16f),
        icon = primary,
    )
}
